test/test_target.py

```diff
@@ -719,6 +720,46 @@ def test_empty_array_stride_check_fortran(ctx_factory):
     knl(queue, input=a_f)
 
 
+def test_passing_bajillions_of_svm_args(ctx_factory):
+    ctx = ctx_factory()
+    queue = cl.CommandQueue(ctx)
+
+    from pyopencl.characterize import has_coarse_grain_buffer_svm
+    if not has_coarse_grain_buffer_svm(queue.device):
+        pytest.skip("device does not support SVM, which is required for this test")
+
+    nargsets = 300
+    knl = lp.make_kernel(
+            "{[i]: 0<=i<n}",
+            [
+                f"c{iargset}[i] = a{iargset}[i]+b{iargset}[i]"
```

Comment:
Adding a gbarrier here makes this test fail:

```suggestion
                f"c{iargset}[i] = a{iargset}[i]+b{iargset}[i]\n...gbarrier\n"
```

```
E       pyopencl._cl.RuntimeError: clBuildProgram failed: BUILD_PROGRAM_FAILURE - clBuildProgram failed: BUILD_PROGRAM_FAILURE - clBuildProgram failed: BUILD_PROGRAM_FAILURE
E
E       Build on <pyopencl.Device 'Tesla K40c' on 'Portable Computing Language' at 0x55ac0cb31100>:
E
E       error: /home/mdiener/.cache/pocl/kcache/tempfile_MlAYH9.cl:6:61: incomplete definition of type 'struct _lpy_arg_struct_loopy_kernel'
E       error: /home/mdiener/.cache/pocl/kcache/tempfile_MlAYH9.cl:7:61: incomplete definition of type 'struct _lpy_arg_struct_loopy_kernel'
E       error: /home/mdiener/.cache/pocl/kcache/tempfile_MlAYH9.cl:8:61: incomplete definition of type 'struct _lpy_arg_struct_loopy_kernel'
[...]
```

The CL source code is missing the declaration of the struct which is used as a function argument in these cases.

Reply:
560b86f56d90014fe441cecf53ea1389def68315 ought to contain a fix.